crates/core/src/protocols/evalcheck/prove.rs

```diff
@@ -372,21 +384,37 @@ where
 		name = "EvalcheckProverState::prove_multilinear",
 		level = "debug"
 	)]
-	fn prove_multilinear(
+	fn prove_multilinear<Challenger_: Challenger>(
 		&mut self,
 		evalcheck_claim: EvalcheckMultilinearClaim<F>,
-	) -> Result<EvalcheckProof<F>, Error> {
-		let EvalcheckMultilinearClaim {
-			id,
-			eval_point,
-			eval,
-		} = evalcheck_claim.clone();
+		transcript: &mut ProverTranscript<Challenger_>,
```

Comment:
I think we can add transcript to the EvalcheckProver and call `self.transcript`  

and it's allow to do something like this 

```
pub fn add_duplicate_claim_to_proof(&mut self, claim_id: usize) {
	serialize_evalcheck_proof(
		&mut self.transcript.message(),
		&EvalcheckProof::DuplicateClaim(claim_id),
	);
}

pub fn add_new_claim_to_proof(&mut self) {
	serialize_evalcheck_proof(&mut self.transcript.message(), &EvalcheckProof::NewClaim);
}
```

Reply:
There's already too many fields on `EvalcheckProver` -- I'd prefer to have it as an explicit argument. It also matches how transcript is used elsewhere in the code.